File: backend/api/analysis.py
from fastapi import APIRouter
from pydantic import BaseModel

from guardrails.output_guards import apply_output_guards
from models.analysis import AnalysisResult
from models.medication import Medication
from tools.interaction_checker import calculate_interaction_matrix

router = APIRouter()


class AnalysisRequest(BaseModel):
    medications: list[Medication]
    session_id: str = ""


@router.post("/full", response_model=AnalysisResult)
async def full_analysis(req: AnalysisRequest) -> AnalysisResult:
    if len(req.medications) < 2:
        return AnalysisResult(
            session_id=req.session_id or None,
            medications=req.medications,
            interactions=[],
        )

    result = await calculate_interaction_matrix(req.medications)
    if req.session_id:
        result.session_id = req.session_id
    result.interactions = apply_output_guards(result.interactions)
    return result

File: frontend/src/api/analysis.ts
import axios from 'axios'
import type { AnalysisResult, Medication } from '@/types'

const api = axios.create({ baseURL: '/api' })

export async function runFullAnalysis(
  medications: Medication[],
  sessionId: string,
): Promise<AnalysisResult> {
  const { data } = await api.post<AnalysisResult>('/analysis/full', {
    medications,
    session_id: sessionId,
  })
  return data
}
